pkg/admissioncontroller/webhook/auth/seed/authorizer.go

```diff
@@ -350,51 +347,123 @@ func (a *authorizer) authorizeServiceAccount(log logr.Logger, seedName string, a
 	return a.authorize(log, seedName, fromType, attrs,
-		[]string{"get", "list", "watch"},
-		nil,
-		nil,
+		withAllowedVerbs("get", "list", "watch"),
 	)
 }
 
+type authzRequest struct {
+	allowedVerbs          []string
+	alwaysAllowedVerbs    []string
+	allowedSubresources   []string
+	listWatchSeedSelector seedSelector
+}
+
+type seedSelector struct {
+	fieldNames []string
+	labelKeys  []string
+}
+
+type configFunc func(req *authzRequest)
+
 func (a *authorizer) authorize(
 	log logr.Logger,
 	seedName string,
 	fromType graph.VertexType,
 	attrs auth.Attributes,
-	allowedVerbs []string,
-	alwaysAllowedVerbs []string,
-	allowedSubresources []string,
+	fns ...configFunc,
 ) (
 	auth.Decision,
 	string,
 	error,
 ) {
-	if ok, reason := a.checkSubresource(log, attrs, allowedSubresources...); !ok {
+	req := &authzRequest{}
+	for _, f := range fns {
+		f(req)
+	}
+
+	if ok, reason := a.checkSubresource(log, attrs, req.allowedSubresources...); !ok {
 		return auth.DecisionNoOpinion, reason, nil
 	}
 
 	// When a new object is created then it doesn't yet exist in the graph, so usually such requests are always allowed
 	// as the 'create case' is typically handled in the SeedRestriction admission handler. Similarly, resources for
 	// which the gardenlet has a controller need to be listed/watched, so those verbs would also be allowed here.
-	if slices.Contains(alwaysAllowedVerbs, attrs.GetVerb()) {
+	if slices.Contains(req.alwaysAllowedVerbs, attrs.GetVerb()) {
 		return auth.DecisionAllow, "", nil
 	}
 
-	if ok, reason := a.checkVerb(log, attrs, append(alwaysAllowedVerbs, allowedVerbs...)...); !ok {
+	if ok, reason := a.checkVerb(log, attrs, append(req.alwaysAllowedVerbs, req.allowedVerbs...)...); !ok {
 		return auth.DecisionNoOpinion, reason, nil
 	}
 
+	if (attrs.GetVerb() == "list" || attrs.GetVerb() == "watch") &&
+		// A resource name is also set when a specific object is read with `.metadata.name` field selector (e.g., in the
+		// single object cache), even for the list verb.
+		// If we have a resource name then we want to consult the graph. There is an exception, though, which is when
+		// the request specifies `.metadata.name` as field name for a seed selector. This means that the client wants
+		// to list/watch the resource with the seed name as field selector. This is a valid scenario and needs to be
+		// handled by the below check function.
+		(attrs.GetName() == "" || slices.Contains(req.listWatchSeedSelector.fieldNames, metav1.ObjectNameField)) {
+		if canAuthorizeWithSelectors, err := a.authorizeWithSelectors.IsPossible(); err != nil {
+			return auth.DecisionNoOpinion, "", fmt.Errorf("failed checking if authorization with selectors is possible: %w", err)
+		} else if canAuthorizeWithSelectors {
+			if ok, reason := a.checkListWatchRequests(log, attrs, seedName, req.listWatchSeedSelector); !ok {
+				return auth.DecisionNoOpinion, reason, nil
```

Comment:
Nit: I would log the message here as this is the path that actually denies the request
```suggestion
			    log.Info("Denying authorization because field/label selectors don't select seed name", "fieldNames", req.listWatchSeedSelector.fieldNames, "labelKeys", req.listWatchSeedSelector.labelKeys)
				return auth.DecisionNoOpinion, reason, nil
```